tests/chainer_tests/training_tests/extensions_tests/test_snapshot.py

```diff
@@ -97,4 +106,160 @@ def exception_raiser(trainer):
+            # uses integer knowledge, timestamp is intentionally
+            # modified here. This comment also applies to other tests
+            # in this file on snapshot freshness.
+            t = base_timestamp + i
+            os.utime(file, (t, t))
+
+        assert self.fmt.format(99) == _find_latest_snapshot(self.fmt,
+                                                            self.path)
+
+
+@testing.parameterize(*testing.product({'fmt':
+                                        ['snapshot_iter_{}_{}',
+                                         'snapshot_iter_{}_{}.npz',
+                                         '{}_snapshot_man_{}-suffix.npz',
+                                         'snapshot_iter_{}.{}']}))
+class TestFindSnapshot2(unittest.TestCase):
+    def setUp(self):
+        self.path = tempfile.mkdtemp()
+        self.files = (self.fmt.format(i*10, j*10) for i, j
+                      in itertools.product(range(0, 10), range(0, 10)))
+
+    def tearDown(self):
+        shutil.rmtree(self.path)
+
+    def test_find_snapshot_files(self):
+        noise = ('tmpsnapshot_iter_{}.{}'.format(i, j)
+                 for i, j in zip(range(10, 304), range(10, 200)))
+
+        for file in itertools.chain(noise, self.files):
+            file = os.path.join(self.path, file)
+            open(file, 'w').close()
+
+        snapshot_files = _find_snapshot_files(self.fmt, self.path)
+
+        answer = [self.fmt.format(i*10, j*10)
+                  for i, j in itertools.product(range(0, 10), range(0, 10))]
+
+        timestamps, snapshot_files = zip(*snapshot_files)
+        answer.sort()
+        snapshot_files = sorted(list(snapshot_files))
+        for lhs, rhs in zip(answer, snapshot_files):
+            assert lhs == rhs
+
+
+@testing.parameterize(*testing.product({'length_retain':
+                                        [(100, 30), (10, 30), (1, 1000),
+                                         (1000, 1), (1, 1), (1, 3), (2, 3)]}))
+class TestFindStaleSnapshot(unittest.TestCase):
+    def setUp(self):
+        self.path = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.path)
+
+    def test_find_stale_snapshot(self):
+        length, retain = self.length_retain
+        fmt = 'snapshot_iter_{}'
+        files = random.sample([fmt.format(i) for i in range(0, length)],
+                              length)
+        base_timestamp = time.time()
+
+        for i, file in enumerate(files):
+            file = os.path.join(self.path, file)
+            open(file, 'w').close()
+
+            # Same comment applies here. See comment in ``TestFindSnapshot``
+            t = base_timestamp + i
+            os.utime(file, (t, t))
+
+        stale = list(_find_stale_snapshots(fmt, self.path, retain))
+        assert max(length-retain, 0) == len(list(stale))
+        stales = [fmt.format(i) for i in range(0, max(length-retain, 0))]
+        for lhs, rhs in zip(stales, stale):
+            lhs == rhs
```

Comment:
`assert` is missing? 
You may as well just write `assert stales == stale`.
Perhaps it's better to rename `stales` to something like `expected`.